cmd/krew/cmd/internal/setup_check_test.go

```diff
@@ -43,6 +43,31 @@ func TestIsBinDirInPATH_secondRun(t *testing.T) {
 	}
 }
 
+// Test case when PATH content is not well normalized
+func TestIsBinDirInPATH_Custom(t *testing.T) {
+	tempDir := testutil.NewTempDir(t)
+	backupPath := os.Getenv("PATH")
+
+	err := os.Setenv("PATH", tempDir.Root()+"/home/../home/user//./.krew/bin")
+	if err != nil {
+		t.Errorf("fail to set PATH with not clean path")
+	}
+	paths := environment.NewPaths(tempDir.Path("/home/../home/user//./.krew"))
+	err = os.MkdirAll(paths.BasePath(), os.ModePerm)
+	if err != nil {
+		t.Errorf("fail to create KREW_ROOT path before IsBinDirInPATH invocation to don4t be on first call")
+	}
+	res := IsBinDirInPATH(paths)
+	if res == false {
+		t.Errorf("expected positive result even if PATH directory are not normalized form")
+	}
+
+	err = os.Setenv("PATH", backupPath)
+	if err != nil {
+		t.Errorf("fail to restore PATH env variable after test")
+	}
```

Comment:
Please move the cleanup into a defer'ed function right after (or before) `os.Setenv` in line 51.